src/modules/builtin/lines.rs

```diff
@@ -42,34 +46,31 @@ impl SyntaxModule<ParserMetadata> for LinesInvocation {
 }
 
 impl TranslateModule for LinesInvocation {
-    fn translate(&self, meta: &mut TranslateMetadata) -> String {
-        let name = format!("__AMBER_ARRAY_{}", meta.gen_value_id());
+    fn translate(&self, meta: &mut TranslateMetadata) -> TranslationFragment {
         let temp = format!("__AMBER_LINE_{}", meta.gen_value_id());
-        let path = (*self.path).as_ref()
-            .map(|p| p.translate_eval(meta, false))
-            .unwrap_or_default();
-        let quote = meta.gen_quote();
-        let dollar = meta.gen_dollar();
+        let path = (*self.path)
+            .as_ref()
+            .map(|p| p.translate(meta))
+            .expect("Cannot read lines without provided path");
         let indent = TranslateMetadata::single_indent();
-        let block = [
-            format!("{name}=()"),
-            format!("while IFS= read -r {temp}; do"),
-            format!("{indent}{name}+=(\"${temp}\")"),
-            format!("done <{path}"),
-        ].join("\n");
-        meta.stmt_queue.push_back(block);
-        format!("{quote}{dollar}{{{name}[@]}}{quote}")
+        let id = meta.gen_value_id();
+        let value = meta.push_stmt_variable("__array", Some(id), Type::array_of(Type::Text), fragments!("()"));
+        meta.stmt_queue.extend([
+            fragments!(raw: "while IFS= read -r {temp}; do"),
+            fragments!(raw: "{indent}{}+=(\"${}\")", value.get_name(), temp),
+            fragments!("done <", path),
+        ]);
+        value.to_frag()
     }
 }
 
 impl LinesInvocation {
-    pub fn surround_iter(&self, meta: &mut TranslateMetadata, name: &str) -> (String, String) {
-        let path = (*self.path).as_ref()
+    pub fn translate_path(&self, meta: &mut TranslateMetadata) -> TranslationFragment {
+        let path = (*self.path)
+            .as_ref()
             .map(|p| p.translate(meta))
-            .unwrap_or_default();
-        let prefix = format!("while IFS= read -r {name}; do");
-        let suffix = format!("done <{path}");
-        (prefix, suffix)
+            .expect("Cannot read lines without provided path in iterator loop");
```

Comment:
Maybe this error can give some details about the loop? like printing the `path`?